lib/Optimizer/Lower.cpp

```diff
@@ -641,6 +641,36 @@ static void lowerTileNode(Function *F, LoweredInfoMap *loweredMap,
   replaceAllUsesOfWith(loweredMap, TN.getResult(), IN);
 }
 
+static void lowerChannelShuffleNode(Function *F, LoweredInfoMap *loweredMap,
+                                    const ChannelShuffleNode &CSN) {
+  auto input = CSN.getInput();
+  auto group = CSN.getGroup();
+  auto kernel = CSN.getKernel();
+
+  auto inDims = input.dims();
+  assert(kernel < inDims.size());
+
+  ShapeVector dims(inDims.begin(), inDims.end());
+  auto D = dims[kernel];
+  assert(D % group == 0);
+
+  dims.erase(dims.begin() + kernel);
+  // Reshape {D1, ... D_k, ... D_n} -> {D1, ... group, D_k / group, ... D_n}
+  dims.insert(dims.begin() + kernel, D / group);
+  dims.insert(dims.begin() + kernel, group);
+  auto *R1 = F->createReshape(CSN.getName().str() + ".reshape1", input, dims);
+
+  std::vector<unsigned_t> transpose(dims.size());
+  for (size_t i = 0; i < transpose.size(); i++)
```

Comment:
nit: I see this was copied from before, but we prefer curly brackets even on single line nests.

Reply:
@jfix71, note that checking the kernel picked up a bug in setting kernel field in `createChannelShuffle` in  `foldChannelShuffle()`.